src/BaseOfTalents/WebUI/Auth/Services/UserAccountService.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Data.Entity.Core;
+using System.Threading.Tasks;
+using DAL.DTO;
+using DAL.Services;
+using Domain.Entities;
+using Mailer;
+using WebUI.Auth.Infrastructure;
+using WebUI.Extensions;
+using WebUI.Globals;
+using WebUI.Globals.Validators;
+using WebUI.Services;
+
+namespace WebUI.Auth.Services
+{
+    using MailsService = BaseService<MailContent, MailDTO>;
+    /// <summary>
+    /// Service for managing user and its session
+    /// </summary>
+    public class UserAccountService : IAccountService
+    {
+        UserService _userService;
+        MailsService _mailService;
+        TemplateService _templateService;
+
+        public UserAccountService(UserService userService, TemplateService templateService,
+            MailsService mailService)
+        {
+            _userService = userService;
+            _mailService = mailService;
+            _templateService = templateService;
+        }
+
+        public UserDTO Register(UserDTO newUser, int mailId)
+        {
+            var mailContent = _mailService.Get(mailId);
+            var template = _templateService.GetTemplate();
+
+            string pwd = PasswordGenerator.GeneratePassword();
+            newUser.Password = (Password)pwd;
```

Comment:
The second option can be implemented. However, this approach seems to be inconvenient in current situation, because original password is needed to be sent, but Password object is an encrypted one